crates/driver/src/domain/competition/auction.rs

```diff
@@ -275,6 +344,12 @@ impl AuctionProcessor {
             }
 
             remaining_balance.0 -= allocated_balance.0;
+
+            // Update order app data if it was fetched.
+            if let Some(fetched_app_data) = app_data_by_hash.remove(&order.app_data.hash()) {
```

Comment:
By returning a mapping `hash -> appdata` we can no longer `.remove()` here since we expect some appdata hashes to be duplicates.